### PR TITLE
Multiple cards

### PR DESCRIPTION
**Pull request by Katerina-codes**
Multiple_cards contains my most recent changes to master branch and development of the program. 
@Maikon @MollieS

### CODE REVIEW

File: lib/blackjack_score.rb [hunk 1]
```diff
@@ -1,24 +1,51 @@
 class BlackjackScore
 
   def get_hand_value(hand)
-    get_card_value(hand[0])
-  end
+    card_values = []
+    hand.each do |card|
```
Comment (felipesere) [inline lib/blackjack_score.rb]: There is a pretty handy (😄) method on `hand` called `map` which maps each `card` to the result of any function you want and return the finished array. 
https://ruby-doc.org/core-2.4.1/Enumerable.html#method-i-map

File: spec/blackjack_score_spec.rb [hunk 2]
```diff
@@ -1,56 +1,74 @@
...
   it "returns 10 if hand is KD" do
-    blackjack_score = BlackjackScore.new
     expect(blackjack_score.get_hand_value(["KD"])).to eq(10)
   end
 
+  it "returns 5 is hand is ['2H', '3D']" do
+    expect(blackjack_score.get_hand_value(["2H", "3D"])).to eq(5)
+  end
+
+  it "returns 5 is hand is ['2H', '3D']" do
```
Comment (felipesere) [inline spec/blackjack_score_spec.rb]: This test is duplicated...

File: spec/blackjack_score_spec.rb [hunk 3]
```diff
@@ -1,56 +1,74 @@
 require 'blackjack_score'
 
 describe BlackjackScore do
+  subject(:blackjack_score) { described_class.new }
 
   it "returns 0 if hand is A" do
-    blackjack_score = BlackjackScore.new
-    expect(blackjack_score.get_hand_value("A")).to eq("Enter a valid card")
+    expect(blackjack_score.get_hand_value(["A"])).to eq(-1)
```
Comment (felipesere) [inline spec/blackjack_score_spec.rb]: This test expects `-1` but the title says `0`.

File: spec/blackjack_score_spec.rb [hunk 4]
```diff
@@ -1,56 +1,74 @@
...
   end
 
   it "returns 0 if hand is C" do
-    blackjack_score = BlackjackScore.new
-    expect(blackjack_score.get_hand_value("C")).to eq("Enter a valid card")
+    expect(blackjack_score.get_hand_value(["C"])).to eq(-1)
```
Comment (felipesere) [inline spec/blackjack_score_spec.rb]: Same as above

File: spec/blackjack_score_spec.rb [hunk 5]
```diff
@@ -1,56 +1,74 @@
...
   end
 
   it "returns 1 if hand is AC" do
-    blackjack_score = BlackjackScore.new
-    expect(blackjack_score.get_hand_value(["AC"])).to eq(1)
+    expect(blackjack_score.get_hand_value(["AC"])).to eq(11)
```
Comment (felipesere) [inline spec/blackjack_score_spec.rb]: This one says it expects 11 but the title says 1

File: spec/blackjack_score_spec.rb [hunk 6]
```diff
@@ -1,56 +1,74 @@
...
   end
 
   it "returns 1 if hand is AS" do
-    blackjack_score = BlackjackScore.new
-    expect(blackjack_score.get_hand_value(["AS"])).to eq(1)
+    expect(blackjack_score.get_hand_value(["AS"])).to eq(11)
```
Comment (felipesere) [inline spec/blackjack_score_spec.rb]: Same as above.

File: lib/blackjack_score.rb [hunk 7]
```diff
@@ -1,24 +1,51 @@
...
+      card.to_i
     end
+  end
+
+  def get_occurence_number(card_values)
+    counts = Hash.new(0)
+    card_values.each do |value|
+      counts[value] += 1
+    end
+    counts[11]
```
Comment (felipesere) [inline lib/blackjack_score.rb]: Why are we returning the 12th element?

File: lib/blackjack_score.rb [hunk 8]
```diff
@@ -1,24 +1,51 @@
...
+    hand.each do |card|
+      value = get_card_value(card)
+      card_values.push(value)
+    end
 
+    sum = card_values.reduce(:+)
+    if sum == -1
+      sum
+    elsif sum > 21 && card_values.include?(11)
+    value_occurences = get_occurence_number(card_values)
```
Comment (felipesere) [inline lib/blackjack_score.rb]: This block is a little confusing to me. Can you extract it to a method and give it an appropriate name?

File: spec/blackjack_score_spec.rb (hunk 3 above)
Comment (andreamazza89) [inline spec/blackjack_score_spec.rb]: I am probably missing knowledge about how blackjack is scored, but the test description seems to be in disagreement with the expectation? (says returns 0 but expects -1)

**Pull request merged**